model/src/main/kotlin/DependencyGraph.kt

```diff
@@ -61,6 +61,39 @@ data class DependencyGraph(
      */
     val scopes: Map<String, List<RootDependencyIndex>>
 ) {
+    companion object {
+        /**
+         * Return a name for the given [scope][scopeName] that is qualified with parts of the identifier of the given
+         * [project]. This is used to ensure that the scope names are unique when constructing a dependency graph from
+         * multiple projects.
+         */
+        fun qualifyScope(project: Project, scopeName: String): String =
+            qualifyScope(project.id, scopeName)
+
+        /**
+         * Return a name for the given [scope][scopeName] that is qualified with parts of the given [projectId]. This
+         * is used to ensure that the scope names are unique when constructing a dependency graph from multiple
+         * projects.
+         */
+        fun qualifyScope(projectId: Identifier, scopeName: String): String =
+            "${projectId.namespace}:${projectId.name}:${projectId.version}:$scopeName"
+
+        /**
+         * Extract the plain (un-qualified) scope name from the given qualified [scopeName]. If the passed in
+         * [scopeName] is not qualified, return it unchanged.
+         */
+        fun unqualifyScope(scopeName: String): String =
+            // To handle the case that the scope contains the separator character, cut off the parts for the
+            // namespace, the name, and the version.
+            scopeName.substringAfterIth(':', 3)
```

Comment:
Can't we simply use `scopeName.substringAfterLast(':')` here (which has a default of `missingDelimiterValue: String = this`)? Or `scopeName.split(':').getOrElse(4) { scopeName }`? I'd really like to get rid of the overly specific `substringAfterIth` function 😁

Reply:
Well, as @mnonnenmacher has correctly pointed out, there is a (small) risk that a scope name contains the separator character. So a robust solution really needs to obtain the substring after the 3rd occurrence of this separator. If I am not missing something, none of the simple solutions does this.

Reply:
Wouldn't passing a limit of 4 to split() cope with that?

Reply:
Yes, this does the trick. Nice.